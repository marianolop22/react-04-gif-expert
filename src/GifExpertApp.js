import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import GifGrid from './components/GifGrid';

// import PropTypes from 'prop-types'

const GifExpertApp = ( {defaultCategories = []} ) => {

    // const [categories, setCategories] = useState(['One Punch']);
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategory={setCategories}/>
            <hr/>
            {/* <button onClick={ handleAdd } >Agregar</button> */}
            <ol>
                {
                    categories.map( (category, index) => 
                        <GifGrid 
                            key={category}
                            category={category}
                        />
                    )
                }
            </ol>
        </>
    );
};

GifExpertApp.propTypes = {
}

export default GifExpertApp
