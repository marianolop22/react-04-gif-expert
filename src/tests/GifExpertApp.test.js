import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from "../GifExpertApp";

describe ('Pruebas de GifExpertApp', ()=> {

    
    test('debe renderizar el GifExpertApp', () => {
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categorias', () => {

        const categories = ['One Punch', 'Saamurai Jack'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        wrapper.find('GifGrid')
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    });


});