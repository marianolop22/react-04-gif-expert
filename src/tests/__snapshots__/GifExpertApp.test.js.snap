// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifExpertApp debe renderizar el GifExpertApp 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
